### PR TITLE
Core Data: Add canRead to useResourcePermissions

### PR DESCRIPTION
**Pull request by adamziel**
## What?

Adds `canRead` property to the `useResourcePermissions` hook. WordPress permissions involve `create`, `update`, `delete`, and `read` actions, and that last one was missing.

## Testing Instructions
Confirm the tests are green

cc @gziolo @Mamaduka

### CODE REVIEW

File: packages/core-data/src/hooks/use-resource-permissions.ts [hunk 1]
```diff
@@ -152,7 +172,7 @@ export default function useResourcePermissions< IdType = void >(
 
 export function __experimentalUseResourcePermissions(
 	resource: string,
-	id?: IdType
+	id?: unknown
```
Comment (gziolo) [inline packages/core-data/src/hooks/use-resource-permissions.ts:175]: Is that change necessary? The type of the stable method didn't change.

**Review by gziolo — APPROVED** [review]
Thank you for adding the missing handling for the `read` permission. Code looks good to me 👍🏻

File: packages/core-data/src/hooks/use-resource-permissions.ts (hunk 1 above)
Comment (adamziel) [reply packages/core-data/src/hooks/use-resource-permissions.ts:175]: The stable method has a generic `IdType` parameter while this one doesn't.

Comment (gziolo) [reply packages/core-data/src/hooks/use-resource-permissions.ts:175]: I really need to learn more about TypeScript 🙈 

Feel free to land this PR 😄
